docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,256 @@
+
+We would ignore amnesia evidence (as individually it's hard to make) and revert to the initial split we had before where `DuplicateVoteEvidence` is also used for light client equivocation attacks and thus we only need `LunaticEvidence`. We would also most likely need to remove `Verify` from the interface as this isn't really something that can be used.
+
+``` go
+type LunaticEvidence struct { // individual lunatic attack
+  header *Header
+  commonHeight int64
+  vote *Vote
+
+  timestamp time.Time // once again taken from the block time at the height of the common header
+}
+```
+
+### Batch Framework
+
+The last approach of this category would be to consider batch only evidence. This works fine with `LightClientAttackEvidence` but would require alterations to `DuplicateVoteEvidence` which would most likely mean that the consensus would send conflicting votes to a buffer in the evidence module which would then wrap all the votes together per height before gossiping them to other nodes and trying to commit it on chain. At a glance this may improve IO and verification speed and perhaps more importantly grouping validators gives the application and Tendermint a better overview of the severity of the attack.
+
+However individual evidence has the advantage that it is easy to check if a node already has that evidence meaning we just need to check hashes to know that we've already verified this evidence before. Batching evidence would imply that each node may have a different combination of duplicate votes which may complicate things.
+
+#### An example of a possible implementation path
+
+`LightClientAttackEvidence` won't change but the evidence interface will need to look like the proposed one above and `DuplicateVoteEvidence` will need to change to encompass multiple double votes. A problem with batch evidence is that it needs to be unique to avoid people from submitting different permutations.
+
+## Decision
+
+Use a hybrid design
+
+We allow individual and batch evidence to coexist together, meaning that verification is done depending on the evidence type and that  the bulk of the work is done in the evidence pool itself (including forming the evidence to be sent to the application).
+
+
+## Detailed Design
+
+Evidence has the following simple interface:
+
+```go
+type Evidence interface {  //proposed
+  Height() int64                                     // height of the offense
+  Bytes() []byte                                     // bytes which comprise the evidence
+  Hash() []byte                                      // hash of the evidence
+  ValidateBasic() error
+  String() string
+}
+```
+
+The changing of the interface is therefore breaking to the block.
+
+We have two concrete types of evidence that fulfil this interface
+
+```go
+type LightClientAttackEvidence struct {
+  ConflictingBlock *LightBlock
+  CommonHeight int64
+}
+```
+where the `Hash()` is the hash of the header and commonHeight
```

Comment:
I would also describe (at a high level) how `ValidateBasic` and `Bytes` are implemented for these two types (`LightClientAttackEvidence` and `DuplicateVoteEvidence`).